transport/transport-runtime/transport-runtime.gradle

```
@@ -13,10 +13,39 @@
 // limitations under the License.
 
 plugins {
+    id "com.google.protobuf"
     id 'firebase-library'
     id 'firebase-vendor'
 }
 
+// add a dependency on the protoc plugin's fat jar to make it available to protobuf below.
+configurations.create("protobuild")
+dependencies {
+    protobuild project(path: ":transport", configuration: "shadow")
+}
+
+protobuf {
+    protoc {
+        artifact = "com.google.protobuf:protoc:$protocVersion"
+    }
+    plugins {
+        firebaseEncoders {
+            path = configurations.protobuild.asPath
+        }
+    }
+    generateProtoTasks {
+        all().each { task ->
+            task.dependsOn configurations.protobuild
+            task.inputs.file 'code-gen-cfg.textproto'
+            task.plugins {
+                firebaseEncoders {
+                    option file('code-gen-cfg.textproto').path
+                }
+            }
+        }
```

You also need to disable the default codegen
```suggestion
        }
        task.builtins {
            remove java
        }
```